lib/fizzy/parser_expr.cpp

```diff
@@ -481,19 +486,23 @@ parser_result<Code> parse_expr(
             if (default_label_idx >= control_stack.size())
                 throw validation_error{"invalid label index"};
 
+            auto& default_branch_frame = control_stack[default_label_idx];
+            const auto default_branch_arity = get_branch_arity(default_branch_frame);
+
             // Remember immediates offset for all br items to fill them at end instruction.
             push(code.immediates, static_cast<uint32_t>(label_indices.size()));
             for (const auto idx : label_indices)
             {
                 auto& branch_frame = control_stack[idx];
-                branch_frame.br_immediate_offsets.push_back(code.immediates.size());
 
+                if (get_branch_arity(branch_frame) != default_branch_arity)
+                    throw validation_error{"br_table labels have inconsistent block result types"};
```

Comment:
```suggestion
                    throw validation_error{"br_table labels have inconsistent types"};
```
... as wabt is reporting this...